lib/pool.js

```diff
@@ -42,24 +42,33 @@ Pool.prototype.request = function (method, path, data) {
       },
       data: data ? JSON.stringify(data) : undefined
     },
-    function (err, res, body) {
-      if (err || Math.floor(res && res.statusCode / 100) !== 2) {
-        var e = new Error(body || (err && err.message))
-        e.statusCode = res && res.statusCode
-        return d.reject(e)
-      }
-      if (!body) { return d.resolve() }
-      var json = null
-      try {
-        json = JSON.parse(body)
-      }
-      catch (e) {
-        return d.reject(new Error('Invalid JSON'))
-      }
-      d.resolve(json)
-    }
+    handleResponse
   )
   return d.promise
+
+  function handleResponse (err, res, body) {
+    var parsedBody = safeParse(body)
+
+    if (err) {
+      return d.reject(err)
+    }
+
+    if (res.statusCode < 200 || res.statusCode >= 300) {
+      var error = parsedBody || new Error(body)
+      error.statusCode = res.statusCode
+      return d.reject(error)
+    }
+
+    if (! body) {
+      return d.resolve()
+    }
+
+    if (parsedBody === body) {
```

Comment:
You're right, that's left over from an earlier version of the function
sorry. :blush: